frontend/src/metabase/visualizations/visualizations/SmartScalar.jsx

```diff
@@ -0,0 +1,192 @@
+import React from "react";
+import { Box, Flex } from "grid-styled";
+import { t, jt } from "c-3po";
+import _ from "underscore";
+
+import { formatNumber, formatValue } from "metabase/lib/formatting";
+import colors from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+import * as Query from "metabase/lib/query/query";
+import * as Card from "metabase/meta/Card";
+import { parseFieldBucketing, formatBucketing } from "metabase/lib/query_time";
+
+import { columnSettings } from "metabase/visualizations/lib/settings/column";
+import {
+  MinRowsError,
+  NoBreakoutError,
+} from "metabase/visualizations/lib/errors";
+
+import ScalarValue, {
+  ScalarWrapper,
+  ScalarTitle,
+} from "metabase/visualizations/components/ScalarValue";
+
+export default class Smart extends React.Component {
+  static uiName = "Smart Scalar";
+  static identifier = "smartscalar";
+  static iconName = "smartscalar";
+
+  static minSize = { width: 3, height: 3 };
+
+  static noHeader = true;
+
+  _scalar: ?HTMLElement;
+
+  static settings = {
+    ...columnSettings({
+      getColumns: ([{ data: { cols } }], settings) => [
+        _.find(cols, col => col.name === settings["scalar.field"]) || cols[1],
+      ],
+    }),
+    "scalar.switch_positive_negative": {
+      title: t`Switch positive / negative colors?`,
+      widget: "toggle",
+    },
+  };
+
+  // Always return true here so that we don't discourage this feature.
+  static isSensible() {
+    return true;
```

Comment:
Since this visualization requires `insights` to be present we should add a check for that.

Currently the full `data` object isn't passed to `isSensible`. I think ideally we'd just change

```
  viz.isSensible(result.data.cols, result.data.rows)
```

in frontend/src/metabase/query_builder/components/VisualizationSettings.jsx

to

```
  viz.isSensible(result.data)
```

and update all the `isSensible` functions in every visualization from

```
  static isSensible(cols, rows) {
```

to 

```
  static isSensible({ cols, rows }) {
```

Then this particular function could just be

```
static isSensible({ insights }) {
  return !!insights;
}
```